tests/Controllers/ContestEntriesControllerTest.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Tests\Controllers;
+
+use App\Models\Contest;
+use App\Models\User;
+use DateTime;
+use Illuminate\Http\Testing\File;
+use Tests\TestCase;
+
+class ContestEntriesControllerTest extends TestCase
+{
+    private Contest $contest;
+    private User $user;
+
+    public function testStoreEntryWithWrongFileExtension()
+    {
+        $this->prepareForStore();
+        $file = File::fake()->create('entry.osz');
+
+        $this
+            ->actingAsVerified($this->user)
+            ->postJson('/community/contest-entries', ['entry' => $file, 'contest_id' => $this->contest->id])
+            ->assertStatus(422);
```

Comment:
this should check error message as well (and add in controller since currently there's none)